### PR TITLE
login page

### PR DESCRIPTION
**Pull request by excellencetechnologies**

### CODE REVIEW

File: actions/todo.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+export const USERS_LOGIN_SUCCESSFULL = "USERS_LOGIN_SUCCESSFULL";
+export const USERS_LOGIN_FAILED = "USERS_LOGIN_FAILED";
+
+import axios from "axios";
+
+function getAsync(username, password) {
+    let data = axios.get("http://localhost:3000/login?username=" + username + "&password=" + password)
```
Comment (upexcel) [inline actions/todo.js]: make a config file and defined this part of url in that 

http://localhost:3000

as a json object

File: containers/addUser.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+function mapStateToProps(state, props) {
+	state = state.toJS()
+	let currentTodo = {
+		username: '',
+		password: '',
+		role: ''
+	};
+	console.log(currentTodo);
```
Comment (upexcel) [inline containers/addUser.jsx]: always remove console.log in pull requests urself

File: containers/login.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    onBack() {
+        this.props.router.push('/admin');
+    }
+    signIn(todo) {
+        this.props.onLogin(todo.username, todo.password).then((val) => {
+            let role = val.data.data.role
+            let err = val.data.error
+            if (role == "admin" && err == 0) {
+                this.props.onLoginSuccess()
+                console.log(val.data.data, "LOGGED IN");
```
Comment (upexcel) [inline containers/login.jsx]: remove

File: containers/login.jsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                this.onBack()
+            } else if (role == "guest" && err == 0) {
+                alert("need to route to user panel");
+            } else {
+                this.props.onLoginFailure(val.data.data)
+                alert(val.data.data)
```
Comment (upexcel) [inline containers/login.jsx]: remove alerts. dont use alerts anywhere. use html to display error as per design

File: index2.jsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import axios from "axios";
+import logger from "redux-logger";
+import thunk from "redux-thunk";
+
+let initialState = Immutable.Map({todos: Immutable.List([])});
+const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
+const middleware = composeEnhancers(applyMiddleware(thunk, logger()));
+let store = createStore(reducer, initialState, middleware);
+
+store.subscribe(() => {
```
Comment (upexcel) [inline index2.jsx]: remove this test code

File: index2.jsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import * as actions from './actions/todo'
+import Immutable from 'immutable';
+import axios from "axios";
+import logger from "redux-logger";
+import thunk from "redux-thunk";
+
+let initialState = Immutable.Map({todos: Immutable.List([])});
+const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
+const middleware = composeEnhancers(applyMiddleware(thunk, logger()));
+let store = createStore(reducer, initialState, middleware);
```
Comment (upexcel) [inline index2.jsx]: define store in another file store/store.js like this and export it from there

File: todo/userslist.jsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+	selectTodo(todo, evt) {
+		this.props.selecttodo(todo);
+	}
+	marktodo(todo, evt) {
+		this.props.marktodo(todo);
+	}
+	render() {
+		var todos = this.props.todos.map((todo) => {
+			return (
+				<TodoItem selectTodo={this.selectTodo} marktodo={this.marktodo} todo={todo} key={todo.id}></TodoItem>
```
Comment (upexcel) [inline todo/userslist.jsx]: what is todoitem doing here?  this component is related to todo?

File: todo/login.jsx [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+			<div>
+				<div className="container">
+					<h2>LOGIN HERE...
+					</h2>
+					<form onSubmit={this.login}>
+						<div className="form-group">
+							<label >Username:</label>
+							<input onChange={() => {
+								this.setState({username: this.refs.username.value})
+							}} type="text" ref="username" value={this.state.username} className="form-control" id="email" placeholder="Enter email"/>
```
Comment (upexcel) [inline todo/login.jsx]: dont use refs

Comment (excellencetechnologies) [inline todo/login.jsx]: that is being used once to access user input on change,and then i m using states everywhere

File: todo/userslist.jsx (hunk 7 above)
Comment (excellencetechnologies) [inline todo/userslist.jsx]: yes, sir i am taking help from todo app list view ,as you viewing userlist and todo list looks almost same,but i will change the name sir

File: index2.jsx (hunk 5 above)
Comment (excellencetechnologies) [inline index2.jsx]: ohk sir

File: index2.jsx (hunk 6 above)
Comment (excellencetechnologies) [inline index2.jsx]: ohk sir

File: containers/login.jsx (hunk 4 above)
Comment (excellencetechnologies) [inline containers/login.jsx]: ohk sir

File: containers/addUser.jsx (hunk 2 above)
Comment (excellencetechnologies) [inline containers/addUser.jsx]: ohk sir

Comment (excellencetechnologies) [inline containers/addUser.jsx]: ohk sir

**Pull request merged**